designs/virtualization/4.6/Next-run-configuration/Next-run-configuration.md

```diff
@@ -0,0 +1,76 @@
+
+## Restart modal
+
+If the user chooses to save and restart the VM they will be asked again to ensure that and will get into a modal including an inline notification that restarting may be disruptive to some logged in users and a notification that the new setting will be applied upon restart. The notification will include a confirmation checkbox so they would be aware of the effect that the restart action might cause and a link to the logged in users list at the bottom of the VM Details page.
+
+![user chooses to resart the VM](img/RestartModal.png)
+
+## VM Details page
+
+In the VM Details page they will be notified that the changes they have made will get applied only after the VM will be restarted.
+Next to the ‘Flavor’ section we will add a link that will lead to the Flavor modal where they can view the pending changes.
+Only in cases when users have made changes to the part of the YAML which are not exposed in the ui, we will add a link to ‘View pending changes in YAML editor’.
+
+![VM details page](img/vm-details.png)
+
+In case there are multiple changes that are on different tabs, we will show a list of them and the same notification will be shown on every tab. Clicking each will take the user to the relevant tab and will open the modal.
+
+![Notification for changes on multiple tabs](img/Notification-changes-on-multiple-tabs.png)
+
+In cases users made changes to the part of the YAML which are not exposed in the ui we’ll add the option to ‘View pending changes in the YAML editor.
+
+![Users can view the changes in the YAML](img/ViewYAML-editor.png)
+
+## List view
+
+In the List view, we will show un-restarted VMs as 'degraded' with yellow warning statuses and 'Pending Changes' subtext underneath the regular Running status. This way they would appear in the Cluster Overview and Project Overview Inventory cards as well. That should make it easier for the users to notice VMs that need to be restarted.
+
+Clicking the 'Running' status will open a popover explaining that this VM has some pending changes that will apply once it will be restarted.
+Users will be able to restart from the popover to save the step of clicking into the kebab menu.
+The popover will also let the user know where to see the pending changes by clicking 'View details' which will take them to the VMs Details page so they can see the 'Restart is required' inline alert at the top.
+
+![Popover indicating a restart is required from the list view](img/ListView.png)
+
+## Adding a ‘Clear changes’ option
+
+Users will be able to view the pending changes from the ‘Edit’ modals and also have the option to clear the changes they have made by clicking the ‘Clear changes’ link.
+
+Clicking 'Flavor' will open the 'Edit Flavor' modal with the inline alert that lets the user clear the change they made.
+
+We will add a (Pending changes) label beside the items that have been changed.
```

Comment:
Maybe instead of labels we can paint the pencil icon in yellow or add a warning sign beside it and have a popup that shows the 'pending changes label'.

Another idea is to have a specific (unique) icon for pending changes warnings and we can put this icon beside each item that has pending changes. WDYT ?

Reply:
I thought about the colored pencil as well but it's not accessible for users with visual limitations. We also don't surface pop up over a pop up.

We use PF4 framework and currently we don't have such an icon, but it is something we might recheck in the future.